src/overlays/actors/ovl_Obj_Etcetera/z_obj_etcetera.c

```diff
@@ -26,35 +32,280 @@ const ActorInit Obj_Etcetera_InitVars = {
+        this->actionFunc = ObjEtcetera_DoIntenseOscillation;
+        Actor_SetScale(&this->dyna.actor, 0.01f);
+        this->dyna.actor.scale.y = 0.02f;
+        this->intenseOscillationScale = 0.003f;
+        this->oscillationTimer = 30;
+        this->burrowFlag &= ~1;
+    } else if ((player->stateFlags3 & 0x2000) && (this->dyna.actor.xzDistToPlayer < 30.0f) &&
+               (this->dyna.actor.yDistToPlayer > 0.0f)) {
+        // Link is hovering above the Deku Flower
+        minOscillationTimer = 10 - (s32)(this->dyna.actor.yDistToPlayer * 0.05f);
+        if (this->oscillationTimer < minOscillationTimer) {
+            this->oscillationTimer = minOscillationTimer;
+        }
+    } else {
+        if (func_800CAF70(&this->dyna)) {
+            if (!(this->burrowFlag & 1)) {
+                // Link is walking onto the Deku Flower, or falling on it from a height
+                this->oscillationTimer = 10;
+                ObjEtcetera_StartSmallFlutterAnimation(this);
+            } else if ((player->actor.speedXZ > 0.1f) || ((player->unk_ABC < 0.0f) && !(player->stateFlags3 & 0x100))) {
+                // Link is walking on top of the Deku Flower, is at the very start of burrowing, or is at the very start
+                // of launching
+                this->oscillationTimer = 10;
+            }
+            this->burrowFlag |= 1;
+        } else {
+            if (this->burrowFlag & 1) {
+                // Link is walking off the Deku Flower
+                this->oscillationTimer = 10;
+                ObjEtcetera_StartSmallFlutterAnimation(this);
+            }
+            this->burrowFlag &= ~1;
+        }
+    }
+    if ((this->collider.base.acFlags & AC_HIT)) {
+        this->oscillationTimer = 10;
+        ObjEtcetera_StartSmallFlutterAnimation(this);
+    }
+    ObjEtcetera_DoNormalOscillation(this, globalCtx);
+}
+
+void ObjEtcetera_PlaySmallFlutterAnimation(ObjEtcetera* this, GlobalContext* globalCtx) {
+    if (func_800CAF70(&this->dyna)) {
+        this->burrowFlag |= 1;
+    } else {
+        this->burrowFlag &= ~1;
+    }
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        this->dyna.actor.draw = ObjEtcetera_DrawIdle;
+        this->actionFunc = ObjEtcetera_Idle;
+    }
+    ObjEtcetera_DoNormalOscillation(this, globalCtx);
+}
+
+void ObjEtcetera_DoIntenseOscillation(ObjEtcetera* this, GlobalContext* globalCtx) {
+    // In order to match, we are seemingly required to access scale.x at one point
+    // without using this. We can create a thisx or dyna pointer to achieve that, but
+    // it's more likely they used dyna given that func_800CAF70 takes a DynaPolyActor.
+    DynaPolyActor* dyna = &this->dyna;
+    f32 scaleTemp;
+
+    if (func_800CAF70(dyna)) {
+        this->burrowFlag |= 1;
+    } else {
+        this->burrowFlag &= ~1;
+    }
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (0 < this->oscillationTimer) {
+        this->oscillationTimer--;
```

Comment:
```suggestion
    if (this->oscillationTimer > 0) {
        this->oscillationTimer--;
```

Reply:
This doesn't match either.

Reply:
Ignore previous comment, this *does* match. For some reason, it was absolutely required to put the 0 first in an earlier version of this function, hence why it was there. However, the proposed change matches now.